wrapper/src/main/java/software/amazon/jdbc/plugin/OpenedConnectionTracker.java

```diff
@@ -87,13 +97,24 @@ public void invalidateAllConnections(final HostSpec hostSpec) {
   }
 
   public void invalidateAllConnections(final String... node) {
-    final Optional<String> instanceEndpoint = Arrays.stream(node).filter(rdsUtils::isRdsInstance).findFirst();
-    if (!instanceEndpoint.isPresent()) {
-      return;
+    TelemetryFactory telemetryFactory = this.pluginService.getTelemetryFactory();
+    TelemetryContext telemetryContext = telemetryFactory.openTelemetryContext(
```

Comment:
Can TelemetryContext inherit AutoClosable to simplify the try-finally blocks?

Reply:
^ It could be a good further improvement. Let's make a ticket for that.

Reply:
We need to look into that. the close() in AutoClosable throws Exception, which means that try-with-resources blocks would have to catch Exception, and we didn't want that in some parts of code. Agree with @sergiyvamz to look at it in a separate ticket